packages/react-scripts/scripts/eject.js

```diff
@@ -36,6 +37,31 @@ prompt(
     process.exit(1);
   }
 
+  // Make sure there are no dirty git status
+  function statusSync() {
+    try {
+      let stdout = execSync(`git status --porcelain`).toString();
+      let status = { dirty: 0, untracked: 0 };
+      stdout.trim().split(/\r?\n/).forEach(file => {
+        if (file.substr(0, 2) === '??') status.untracked++;
+        else status.dirty++;
+      });
+      return status;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  const status = statusSync();
+  if (status && status.dirty) {
+    console.error(
+      `Git repository has ${status.dirty} dirty ${status.dirty > 1 ? 'files' : 'file'}. ` +
+        'We cannot continue as you would lose all the changes in that file or directory. ' +
+        'Please push commit before and run this command again.'
```

Comment:
Thanks for those suggestions @sunjay, @taylorlapeyre. I personally liked @taylorlapeyre version. I will make an update soon.